src/main_window.py

```diff
@@ -1612,7 +1613,11 @@ def logout(self, action, param):
                 os.system("dbus-send --print-reply --session --dest=org.kde.LogoutPrompt /LogoutPrompt org.kde.LogoutPrompt.promptLogout")
             elif self.win.environment == 'COSMIC (New)':
                 os.system("dbus-send --print-reply --session --dest=com.system76.CosmicSession --type=method_call /com/system76/CosmicSession com.system76.CosmicSession.Exit")
+            elif self.win.environment == 'Hyprland':
+                os.system("hyprctl dispatch exit") #does logout without prompting the user because i couldn't find a good way to do it
```

Comment:
```suggestion
                os.system("hyprctl dispatch exit") if not flatpak else os.system(f'notify-send "{_["err_occured"]}" "It is not possible to log out of the system in the Flatpak environment. Please log out manually."')
```
Unfortunately I also don't know what D-Bus interface to use for Hyprland, so I've set it up so that if the application is running outside of Flatpak, the command is executed, if it's not, a notification is sent that you need to log out manually.

Reply:
i found a way to bypass the dbus and just kill the process but will need to test if it works it is with psutil os and signal

Reply:
unfortunately i can not kill or send a dbus req to hyprland since it doesn't expose a port/bus so i think ill just look into how to remove that button

Reply:
Instead of importing the `psutil` and `signal` modules, you can use this command, which should hopefully work on all DEs (I tried it on GNOME, and it works): 
```bash
dbus-send --system --print-reply --dest=org.freedesktop.login1 /org/freedesktop/login1 'org.freedesktop.login1.Manager.TerminateSession' string:$(dbus-send --system --print-reply --dest=org.freedesktop.login1 /org/freedesktop/login1 'org.freedesktop.login1.Manager.ListSessions' | awk -F 'string "' '/string "/ {print $2; exit}' | awk -F '"' '{print $1}')
```
however, it is necessary to add permissions to the Flatpak manifest in the finish-args section: `--system-talk-name=org.freedesktop.login1`.